c10/cuda/driver_api.cpp

```diff
@@ -47,6 +43,32 @@ C10_EXPORT DriverAPI* DriverAPI::get() {
   return &singleton;
 }
 
+void* get_symbol(const char* name) {
+  // CUDA 12.5+ supports version-based lookup
+#if defined(CUDA_VERSION) && (CUDA_VERSION >= 12050)
+  // We hardcode the version to 12.0 to ensure we always use a consistent
+  // version of the driver API.
+  unsigned int req_ver = 12000;
+  if (auto st = cudaGetDriverEntryPointByVersion(
```

Comment:
how will this work to return symbols that are added after 12000, like `cuMulticastAddDevice`?

Reply:
Originally, I thought we should use the fallback API `cudaGetDriverEntryPoint`. However, I believe @wujingyue intended to remove all uses of `cudaGetDriverEntryPoint` and replace them with `cudaGetDriverEntryPointByVersion`, specifying a version number for each API.

`cudaGetDriverEntryPoint` is still kept as a backup for now, and we resolve symbols using `cudaGetDriverEntryPointByVersion` whenever possible.

Reply:
> how will this work to return symbols that are added after 12000, like `cuMulticastAddDevice`?

In that case, the requested version should be the version that introduced cuMulticastAddDevice, which I believe is 12090. Any use of cuMulticastAddDevice has to be version guarded (e.g. `#if (CUDA_VERSION >= 12090)`) as needed anyway.